### PR TITLE
Change read more link for warning message on Apple Pay domain registration failure

### LINKED ISSUE #4511: Apple Pay: fix errors on web registrations
### Describe the bug

When a domain is registered, we saw a burst of domain registration errors.
Possible causes: 
1. `apple_pay_bad_status_code ` - meaning when `<domain>/.well-known/apple-developer-merchantid-domain-association` is accessed, we get a non 200 HTTP response
2. `apple_pay_unsupported_country` - meaning the merchant is in an unsupported country. Supported countries: https://support.stripe.com/questions/country-availability-for-apple-pay 

Context: p1657312179990539-slack-C9976E5MJ

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #4511

#### Changes proposed in this Pull Request
When Apple Pay domain verification fails, an error notification is shown with a "Learn more" link that points to the docs. We need to change the docs link, to point to the domain registration steps ( https://woocommerce.com/document/woocommerce-payments/payment-methods/apple-pay/#domain-registration ) 

#### Testing instructions

In order to register the domain to Apple Pay, server must use the WCPay live restricted keys for Apple Pay. We can use the account `acct_...ZoSU` . You can find the account here: paJDYF-18Y-p2#comment-5116.
Dev mode must be disabled in WCPay dev utils.
In order to trigger the domain registration, we must simulate that the plugin was just updated.
* change the content of the `apple-developer-merchantid-domain-association` 
* run the following SQL to simulate that you are on an older version: `UPDATE wp_options SET option_value='3.1.0' WHERE option_name='woocommerce_woocommerce_payments_version';`
* Refresh the admin page. You should see the following error:
Screenshot 2023-01-11 at 16 17 53
* Check the "Learn more" link
* Refresh the page again. You should see the following error:
Screenshot 2023-01-11 at 16 18 08
* Check the "Learn more" link
* If you want to run the test again, change the `docker/wordpress/.well-known/apple-developer-merchantid-domain-association` with its original value in order to be different from the one from root directory.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: changelog/fix-4511-change-registration-domain-docs-link [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Change Learn More link for the error notification when Apple Pay domain registration failed
```
Comment (anu-rock) [inline changelog/fix-4511-change-registration-domain-docs-link]: We can make it a bit more readable:

Point the "Learn more" link to a more appropriate document in the Apple Pay domain registration failure notification.

**Review by anu-rock — COMMENTED** [review]
Code-only review: LGTM based on our earlier discussion. I've left a minor comment.

**Review by ricardo — APPROVED** [review]
LGTM

**Review by elazzabi — APPROVED** [review]
This looks good! Reviewed the code only as I don't think it needs a manual test.

**Pull request merged**